tap_hubspot/marketing_streams.py

```diff
@@ -0,0 +1,167 @@
+"""Stream type classes for tap-hubspot."""
+# from black import Report
+from asyncio.log import logger
+from math import inf
+import requests
+import singer
+import json
+
+from dateutil import parser
+import datetime, pytz
+import time
+from datetime import datetime
+
+from pathlib import Path
+from typing import Any, Dict, Optional, Union, List, Iterable
+
+from singer_sdk import typing as th  # JSON Schema typing helpers
+from pathlib import Path
+from typing import Any, Dict, Optional, Union, List, Iterable
+
+from memoization import cached
+
+from singer_sdk.helpers.jsonpath import extract_jsonpath
+from singer_sdk.streams import RESTStream
+from singer_sdk.authenticators import BearerTokenAuthenticator
+from singer_sdk import typing as th  # JSON schema typing helpers
+from tap_hubspot.client import HubspotStream
+from tap_hubspot.schemas.marketing import CampaignIds
+
+SCHEMAS_DIR = Path(__file__).parent / Path("./schemas")
+
+LOGGER = singer.get_logger()
+utc=pytz.UTC
+
+from tap_hubspot.schemas.marketing import (
+    Emails,
+    CampaignIds,
+    Campaigns,
+    Forms
+)
+
+class MarketingStream(HubspotStream):
+    records_jsonpath = "$.results[*]"  # Or override `parse_response`.
+    next_page_token_jsonpath = "$.paging.next.after"  # Or override `get_next_page_token`.
+    replication_key = "updatedAt"
+    replication_method = "INCREMENTAL"
+    cached_schema = None
+    properties = []
+    schema_filepath = ""
+
+
+class MarketingEmailsStream(MarketingStream):
+    records_jsonpath = "$.objects[*]"  # Or override `parse_response`.
+    next_page_token_jsonpath = "$.offset"  # Or override `get_next_page_token`.
+    version = "v1"
+    name = "marketing_emails_v1"
+    path = f"/marketing-emails/{version}/emails/with-statistics"
+    primary_keys = ["id"]
+    replication_key = "updated"
+    total_emails = inf
+
+    def parse_response(self, response: requests.Response) -> Iterable[dict]:
+        """Parse the response and return an iterator of result rows."""
+        self.total_emails = response.json()['total']
+        data = response.json()
+        ret = []
+        for d in data["objects"]:
+            val = d
+            val["updated"] = datetime.fromtimestamp(d["updated"]/1000, tz=utc)
+            ret.append(val)
```

Comment:
Same as I commented on another PR, I think this can be cleaner :)
```suggestion
        ret = [dict(d, updated=datetime.fromtimestamp(d["updated"]/1000, tz=utc)) for d in data["objects"]
```

Reply:
Beautiful, I'll be sure to use that; sorry I'm going rapid fire on some commits